bsdlib/doc/ug_bsdlib_porting_os.rst

```diff
@@ -0,0 +1,292 @@
+
+*Required action*:
+
+* Clear the low priority BSD library scheduling IRQ using OS primitives or NVIC functions.
+
+bsd_os_application_irq_set
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This function is called by the BSD library when the library wants to set a pending IRQ on the low priority BSD library scheduling IRQ.
+
+*Required action*:
+
+* Set a pending IRQ on the low priority BSD library scheduling IRQ using OS primitives or NVIC functions.
+
+bsd_os_trace_irq_clear
+~~~~~~~~~~~~~~~~~~~~~~
+
+This function is called by the BSD library when the library wants to clear IRQ on the low priority trace scheduling IRQ.
+
+*Required action*:
+
+* Clear the low priority trace scheduling IRQ using OS primitives or NVIC functions.
+
+bsd_os_trace_irq_set
+~~~~~~~~~~~~~~~~~~~~
+
+This function is called by the BSD library when the library wants to set a pending IRQ on the low priority trace scheduling IRQ.
+
+*Required action*:
+
+* Set a pending IRQ on the low priority trace scheduling IRQ using OS primitives or NVIC functions.
+
+bsd_os_trace_put
+~~~~~~~~~~~~~~~~
+
+This function puts the trace string to the desired medium, typically UART.
+However, the medium used to forward and store the traces is up to the implementation and must be initialized correctly before used.
+If you are not interested in traces, they can be ignored and this function can be empty and simply return.
+
+bsd_os_application_irq_handler
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This function is implemented in the BSD library and must be called upon the low priority BSD library IRQ handler, triggered by the :cpp:func:`bsd_os_application_irq_set` function.
+
+bsd_os_trace_irq_handler
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This function is implemented in the BSD library and must be called upon the low priority trace IRQ handler, triggered by the :cpp:func:`bsd_os_trace_irq_set` function.
+
+Other scenarios to handle in bsd_os.c
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+#. In case the OS has its own IRQ handler scheme, not directly forwarding the IPC_IRQHandler to the BSD library, this must be routed by the OS.
+   The OS must call IPC_IRQHandler() upon all IRQs with IRQ number IPC_IRQn.
+#. In :file:`bsd_os.c`, you can configure a desired medium for forwarding trace data upon :cpp:func:`bsd_os_trace_put` calls .
+
+Memory
+******
+
+The BSD library must have a region of absolute positioned RAM.
+The region must be configured as non-secure RAM by the secure application.
+The starting address of this region must be 0x20010000, and the size of the block is 0xC020 bytes.
+
+The secure domain memory is divided into segments of 8 KB RAM blocks.
+Because the size of the BSD library block is 0xC020 bytes, it is not aligned to full 8 KB blocks.
+It occupies six blocks and a small part of a seventh block.
+
+The leftover memory in that seventh block (8160 bytes, starting at address 0x2001 c020) is configured as non-secure, but it can be used by the application.
```

Comment:
```suggestion
The leftover memory in that seventh block (8160 bytes, starting at address 0x2001C020) is configured as non-secure, but it can be used by the application.
```